presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -238,10 +243,17 @@ protected PlanNodeStatsEstimate visitComparisonExpression(ComparisonExpression n
                 return process(new ComparisonExpression(type.flip(), right, left));
             }
 
-            if (left instanceof SymbolReference && right instanceof Literal) {
-                Symbol symbol = Symbol.from(left);
-                OptionalDouble literal = doubleValueFromLiteral(types.get(symbol), (Literal) right);
-                return comparisonSymbolToLiteralStats(input, symbol, literal, type);
+            if (left instanceof Literal && !(right instanceof Literal)) {
```

Comment:
it looks like it also has to take care of casts

Reply:
I don't understand.